core/src/main/java/com/mongodb/hadoop/splitter/StandaloneMongoSplitter.java

```diff
@@ -164,17 +168,28 @@ public StandaloneMongoSplitter(final Configuration conf) {
                 LOG.warn("WARNING: No Input Splits were calculated by the split code. Proceeding with a *single* split. Data may be too"
                          + " small, try lowering 'mongo.input.split_size' if this is undesirable.");
             }
-
+            
             BasicDBObject lastKey = null; // Lower boundary of the first min split
-
+            //splitKey is BSONObject of shape { key: 1 }
+            
+            String splitKeyString = (new BasicDBObject(splitKey.toMap())).keySet().iterator().next();
+            //If splitKeyMin has been used, use it as first boundary.
+            if(splitKeyMin.containsField(splitKeyString)) {
```

Comment:
Of course I can simplify the if-condition. But still i need to keep the line:
    lastKey = new BasicDBObject(splitKeyMin.toMap());
because of createSplitFromBounds expecting a BasicDBObject.
But:
    String splitKeyString = (new BasicDBObject(splitKey.toMap())).keySet().iterator().next();
can be removed.